Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -1235,11 +1235,10 @@ class BrowserViewController: UIViewController {
     }
 
     private func getCurrentUIState() -> UIState {
-        guard let tab = tabManager.selectedTab,
-        let displayURL = tab.displayURL where displayURL.absoluteString.characters.count > 0 else {
-            if let homePanelController = homePanelController {
-                return .HomePanels(homePanelState: homePanelController.homePanelState)
-            }
+        if let homePanelController = homePanelController {
+            return .HomePanels(homePanelState: homePanelController.homePanelState)
+        }
+        guard let tab = tabManager.selectedTab where !tab.loading else {
```

Comment:
The only likely race could come from https://github.com/mozilla/firefox-ios/blob/76c60f58b9cdf6da65397d819c9cdaf555632b11/Client/Frontend/Browser/BrowserViewController.swift#L747 .

Merging, given https://bugzilla.mozilla.org/show_bug.cgi?id=1281403#c2